geopandas/tools/overlay.py

```diff
@@ -176,3 +177,158 @@ def overlay(df1, df2, how, use_sindex=True):
 
     # Return geodataframe with new indices
     return GeoDataFrame(collection, index=range(len(collection)))
+
+def overlay_intersection(df1, df2):
+    ''''
+    Overlay Intersection operation used in overlay function
+    '''
+    # Spatial Index to create intersections
+    spatial_index = df2.sindex
+    df1['bbox'] = df1.geometry.apply(lambda x: x.bounds)
+    df1['sidx'] = df1.bbox.apply(lambda x:list(spatial_index.intersection(x)))
+    # Create pairs of geometries in both dataframes to be intersected
+    pairs = df1['sidx'].to_dict()
```

Comment:
You can also iterate over `df1['sidx'].iteritems()` instead of first converting to dict and iterating over items. That will be slightly faster (although probably negligible compared to the rest of the algo), but also slightly cleaner code-wise.